OutOfSchool/OutOfSchool.WebApi/Services/ApplicationService.cs

```diff
@@ -90,6 +91,14 @@ public class ApplicationService : IApplicationService, INotificationReciever
 
             ModelNullValidation(applicationDto);
 
+            var workshopStatus = await CheckWorkshopStatus(applicationDto.WorkshopId).ConfigureAwait(false);
+
+            if (workshopStatus == WorkshopStatus.Closed)
+            {
+                logger.LogInformation("Unable to create a new application for a workshop because workshop status is closed.");
+                throw new ArgumentException("Unable to create a new application for a workshop because workshop status is closed.");
+            }
```

Comment:
`CheckWorkshopStatus` does not perform the status checking. You should probably move this piece of code inside the `CheckWorkshopStatus` method.